### ISSUE TITLE
"wiggles" in wire path

### ISSUE THREAD

**Issue by andrejsho** [Philipp-MR/CoilGen#10]
Hi,
thanks a lot for the CoilGen software, it helps me a lot for a project in my physics master.
I am building gradient coils for a halbach magnet MRI system. As of now I am struggling with the "wiggles" that the wirepath has sometimes on one side of the coil (in the attached photo on the left the wirepath is smooth, on the right there are wiggles). Since I plan to 3d print a holder with grooves for the wire, I would like to have the wirepath smoother. 
I tried to increase the smooth factor, but that did not improve it much, also when I increase the smooth factor to 3, then I get an out of memory error (my laptop has 16gb RAM, and I have also attached the code I am using to calculate the coils). 
Is there a way to reduce these wiggles?
Thanks in advance!
Andrej

[generate_gradients.zip](https://github.com/user-attachments/files/18029188/generate_gradients.zip)

![zGradient](https://github.com/user-attachments/assets/f338f590-ff07-4026-a420-5baa592aa8a8)

**Comment by Philipp-MR** [comment]
Hi Andrej,

Great to hear that you like my software. The problem you mention is familiar to me. It is related to the mesh resolution of you clindrical surface. The first measure would be to increase the mesh resolution. You can also try to increase the smoothing parameters. Look for them in the code.
It should be possible to solve you problem!

Best,
Philipp

**Comment by andrejsho** [comment]
Hi Philipp,
thank you for your help. Do I understand correctly that I set the mesh resolution in 'cylinder_mesh_parameter_list' as third and fourth parameter? Increasing those did not help much, up to the point that my laptop ran out of memory during calculations (same holds for smooth factor).
However, even though I am not very experienced in Python I tried the pyCoilGen software in the last tow days and there I could increase the smooth factor up to 4 or 5 without memory issues. This made the paths smooth. 
Kind regards,
Andrej
